WalletWasabi/Services/CcjCoordinator.cs

```diff
@@ -433,9 +433,9 @@ public bool AnyRunningRoundContainsInput(OutPoint input, out List<Alice> alices)
 			}
 		}
 
-		public bool ContainsCoinJoin(uint256 hash)
+		public async Task<bool> ContainsCoinJoinAsync(uint256 hash)
 		{
-			using (CoinJoinsLock.Lock())
+			using (await CoinJoinsLock.LockAsync())
```

Comment:
It could worth to replace the `List<uint256>` by an `ImmutableList<uint256>` in order to eliminate the need for these locks and async locks.

Reply:
That doesn't eliminate the need for locks, similarly to the Concurrent Collections. Locks make sure that the data is consistent with the business logic. immutable/concurrent collections simply don't throw exceptions when writing or reading them from different threads.

Reply:
ok